src/online2bin/online2-tcp-nnet3-decode-faster.cc

```diff
@@ -0,0 +1,457 @@
+  GetLinearSymbolSequence(lat, &alignment, &words, &weight);
+
+  std::string msg;
+  for (size_t i = 0; i < words.size(); i++) {
+    std::string s = word_syms->Find(words[i]);
+    if (s.empty()) {
+      KALDI_ERR << "Word-id " << words[i] << " not in symbol table.";
+      msg += "<#" + std::to_string(i) + "> ";
+    } else
+      msg += s + " ";
+  }
+  return msg;
+}
+
+std::string LatticeToString(const CompactLattice &clat, fst::SymbolTable *word_syms) {
+  if (clat.NumStates() == 0) {
+    KALDI_WARN << "Empty lattice.";
+    return "";
+  }
+  CompactLattice best_path_clat;
+  CompactLatticeShortestPath(clat, &best_path_clat);
+
+  Lattice best_path_lat;
+  ConvertLattice(best_path_clat, &best_path_lat);
+  return LatticeToString(best_path_lat, word_syms);
+}
+}
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+    using namespace fst;
+
+    typedef kaldi::int32 int32;
+    typedef kaldi::int64 int64;
+
+    const char *usage =
+        "Reads in audio from a network socket and performs online decoding with neural nets\n"
+        "(nnet3 setup), with iVector-based speaker adaptation and\n"
+        "endpointing.  Note: some configuration values and inputs are\n"
+        "set via config files whose filenames are passed as options\n"
+        "\n"
+        "Usage: online2-net-nnet3-latgen-faster [options] <nnet3-in> <fst-in> "
+        "<word-symbol-table> <listen-port>\n";
+
+    ParseOptions po(usage);
+
+
+    // feature_opts includes configuration for the iVector adaptation,
+    // as well as the basic features.
+    OnlineNnet2FeaturePipelineConfig feature_opts;
+    nnet3::NnetSimpleLoopedComputationOptions decodable_opts;
+    LatticeFasterDecoderConfig decoder_opts;
+    OnlineEndpointConfig endpoint_opts;
+
+    BaseFloat chunk_length_secs = 0.18;
+    BaseFloat output_freq = 1;
+    BaseFloat samp_freq = 16000.0;
+    int read_timeout = 3;
+
+    po.Register("samp-freq", &samp_freq,
+                "Sampling frequency of the input signal (coded as 16-bit slinear).");
+    po.Register("chunk-length", &chunk_length_secs,
+                "Length of chunk size in seconds, that we process.");
+    po.Register("output-freq", &output_freq,
+                "How often in seconds, do we check for changes in output.");
+    po.Register("num-threads-startup", &g_num_threads,
+                "Number of threads used when initializing iVector extractor.");
+    po.Register("read-timeout", &read_timeout,
+                "Number of seconds of timout for TCP audio data to appear on the stream. Use -1 for blocking.");
+
+    feature_opts.Register(&po);
+    decodable_opts.Register(&po);
+    decoder_opts.Register(&po);
+    endpoint_opts.Register(&po);
+
+    po.Read(argc, argv);
+
+    if (po.NumArgs() != 4) {
+      po.PrintUsage();
+      return 1;
+    }
+
+    std::string nnet3_rxfilename = po.GetArg(1),
+        fst_rxfilename = po.GetArg(2),
+        word_syms_rxfilename = po.GetArg(3),
```

Comment:
word_syms cannot be read from an rxfilename, it need to be regular filename.

word_syms_rxfilename -> word_syms_filename